src/web/app/components/question-responses/single-statistics/single-statistics.component.ts

```diff
@@ -62,6 +63,11 @@ export class SingleStatisticsComponent implements OnInit, OnChanges {
       return this.feedbackResponsesService
           .isFeedbackResponsesDisplayedOnSection(response, this.section, this.sectionType);
     });
+
+    this.responses.forEach((response: ResponseOutput) => {
+      response.giver = StringHelper.removeAnonymousHash(response.giver);
+      response.recipient = StringHelper.removeAnonymousHash(response.recipient);
+    });
```

Comment:
I don't think this logic should be handled in `filterResponses`.
You can extract out this part of the code into a new method and call it within `ngOnChanges` and `ngOnInit` instead.